helix-view/src/editor.rs

```diff
@@ -239,6 +242,13 @@ impl Editor {
         (view, doc)
     }
 
+    pub fn current_with_registers(&mut self) -> (&mut View, &mut Document, &mut Registers) {
+        let view = self.tree.get_mut(self.tree.focus);
+        let doc = &mut self.documents[view.doc];
```

Comment:
```suggestion
        let (view, doc) = self.current();
```

Reply:
This won't work because `self.current()` mutably borrows self, so then `self.registers` fails to borrow. We need to change `current` to a macro to get around this